docs/arch/adr-3-store-status.md

```diff
@@ -0,0 +1,123 @@
+|Claim Review                 |Unpublished changes|In Review           |
+|2i Approved                  |Unpublished changes|2i Approved         |
+|2i Requests changes          |Unpublished changes|N/A                 |
+|Publish                      |Published          |N/A                 |
+|Schedule                     |Unpublished changes|Scheduled           |
+|Unschedule                   |Unpublished changes|2i Approved         |
+|Discard Draft                |Published          |N/A                 |
+|Unpublish                    |Draft              |N/A                 |
+
+
+### Status assumptions
+
+- Can only publish if workflow state is 2i Approved
+- Can only schedule if workflow state is 2i Approved
+- Can only Claim Review if workflow state is Submitted for 2i and the Claimer is not the same user as the one that submitted for 2i
+- After 2i has been approved, the user can submit for 2i again
+- If the 2i reviewer requests changes, the workflow state is reset and the 2i workflow starts again
+- Can only delete a step-by-step if it’s never been published
+- "live" will be renamed to "published" to be consistent with the other publishing apps.
+
+
+## Decision
+
+Create a new field in the database to store a value of `status` that allows values of:
+
+- draft
+- published
+- scheduled
+- submitted_for_2i
+- in_review
+- 2i_approved
+
+
+"draft" and "unpublished changes" mean the same thing in regards to their position in the publishing workflow, so if they are consolidated it won't be necessary to keep track of separate publication and workflow states. The presence of a `published_at` date can be used to determine if something has been previously published, as it is now.
+
+That means the 2i workflow can be simplified.
+
+### 2i workflow
+
+|Action                                  |Starting Status |End Status          |
+|----------------------------------------|:--------------:|-------------------:|
+|Create new/update published step-by-step|N/A or published|draft               |
+|Makes changes to step-by-step           |draft           |draft               |
+|Submit for 2i                           |draft           |submitted_for_2i    |
+|Claim Review                            |submitted_for_2i|in_review           |
+|2i Approved                             |in_review       |2i_approved         |
+|2i Requests changes                     |in_review       |draft               |
+|Publish                                 |2i_approved     |published           |
+|Schedule                                |2i_approved     |scheduled           |
+|Unschedule                              |scheduled       |2i_approved         |
+
+
+### Other actions
+
+- Delete: Only if step-by-step is not scheduled or been previously published
+- Discard Draft: Only if step-by-step is not scheduled and has been previously published
+- Unpublish: Only if step-by-step is not scheduled and has been previously published
```

Comment:
This might be out of scope for this, but `unpublish` might have several options too:
- unpublish and 404 (`vanish`?)
- unpublish with redirect
- unpublish with 410

It might be useful to store these statuses too, so that it's obvious what's happened

Reply:
At the moment we only allow unpublish with redirect. It's definitely something we need to look at further though.